### DISCUSSION TITLE
'required' doesn't work as expected

### DISCUSSION THREAD

**Discussion by dor272** [typegoose/typegoose#467] [Q&A] [answered]
Hi,

 I have a model that has a 2 required references to another model, both references are of the same type:

```ts
class X {
  @prop({required: true, ref: 'Y'})
   public a!: Ref<Yclass>
  @prop({required: true, ref: 'Y'})
   public b!: Ref<Yclass>
}
```

Now I started to write unit test for it, when none of `a` or `b` provided and I try to validate the model I get errors for missing `a` and `b` as I should.. (`new X({})`)

When I init X with just 1 of them:
`new X({a: ObjectId})` is passes validation even though `b` is missing.

Am I doing something wrong or is it a bug?

Thanks.

**Comment by hasezoey** [comment, marked as answer]
what are the versions of the following do you have: `typescript typegoose mongoose @types/mongoose`?

and if you mean runtime, then typegoose has not really anything to do with it

**Reply by dor272** [reply]
`typescript: 4.1.2`
`typegoose: 7.4.5`
`mongoose: 5.10.18`
`@types/mongoose: 5.10.3`

typegoose converts the class into a mongoose model, so maybe something goes wrong there...
Both `a` and `b` are required but once 1 provided to the model it acts as if both were provided.

**Reply by hasezoey** [reply]
i tested it, all works as it should, used code:

<details>

```ts
// NodeJS: 14.15.3
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop, Ref } from "@typegoose/typegoose"; // @typegoose/typegoose@7.4.6
import * as mongoose from "mongoose"; // mongoose@5.10.18 @types/mongoose@5.10.3

class Y {
  @prop()
  public dummy?: string;
}

const YModel = getModelForClass(Y);

class X {
  @prop({ required: true, ref: 'Y' })
  public a!: Ref<Y>;
  @prop({ required: true, ref: 'Y' })
  public b!: Ref<Y>;
}
const XModel = getModelForClass(X);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verify467", useCreateIndex: true, useUnifiedTopology: true });

  // fails
  // {
  //   const y1 = new YModel({ dummy: "hi1" });
  //   const x1 = new XModel({});

  //   await x1.validate(); // should fail, because both "a" and "b" are missing
  // }

  // fails
  // {
  //   const y1 = new YModel({ dummy: "hi2" });
  //   const x1 = new XModel({ a: y1 });

  //   await x1.validate(); // should fail, because "b" is missing
  // }

  // fails
  // {
  //   const y1 = new YModel({ dummy: "hi3" });
  //   const x1 = new XModel({ b: y1 });

  //   await x1.validate(); // should fail, because "a" is missing
  // }

  // runs
  {
    const y1 = new YModel({ dummy: "hi4" });
    const y2 = new YModel({ dummy: "hi5" });
    const x1 = new XModel({ a: y1, b: y2 });

    await x1.validate(); // should not fail
  }

  await mongoose.disconnect();
})();
```

Full Project: https://github.com/typegoose/typegoose-testing/tree/verify467
</details>
